examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,366 @@
+    subprocs = []
+    for rank in range(num_tasks):
+        sp = pool.apply_async(train_finetune, (finetune_tasks[rank], save_ckpts[rank], ckpt_load_path, wandb_group_name, save_folder, save_locally, wandb_logger, rank, bucket, load_ignore_keys), callback=log_metrics)
+        subprocs.append(sp)
+    for sp in subprocs:
+        sp.wait()
+    pool.close()
+    pool.join() # wait for join to proceed
+   
+    cuda_envs.close() 
+    cuda_envs.join_thread()
+    
+'''Run single instance of a finetuning job on given task. ''' 
+def train_finetune(task: str, save_ckpt: bool, load_path: str, wandb_group_name: str, save_folder: str, save_locally: bool, wandb_logger: Union[WandBLogger, None], gpu_id: int, bucket: Optional[str] = None, load_ignore_keys: Optional[List[str]] = None):
+    ft_hparams = TrainerHparams.create(cli_args=False, f=f'./composer/yamls/models/glue/{task}.yaml') 
+    ft_hparams.load_path = load_path
+    ft_hparams.device = DeviceGPU(gpu_id)
+    ft_hparams.log_to_console = True
+    ft_hparams.progress_bar = False
+    ft_hparams.load_ignore_keys = load_ignore_keys
+    
+    # add finetune-specific tags to the logger 
+    if wandb_logger:
+        wandb_logger._init_kwargs['tags'] = ['finetune', task]
+        wandb_logger._init_kwargs['group'] = wandb_group_name
+        wandb_logger._rank_zero_only = True
+        if not ft_hparams.loggers:
+            ft_hparams.loggers = wandb_logger
+        else:
+            ft_hparams.loggers.append(wandb_logger)
+
+    # load checkpoint to finetune on via cloud 
+    if not save_locally:
+        ft_hparams.load_object_store = S3ObjectStoreHparams(bucket=bucket)
+
+    # saving single checkpoint at the end of training the task 
+    if save_ckpt:
+        if save_locally:
+            task_ckpt_path = os.path.join(save_folder, f'{task}')
+            ft_hparams.save_folder = task_ckpt_path
+            ft_hparams.save_num_checkpoints_to_keep = 1
+            ft_hparams.save_overwrite = True
+
+            if not os.path.exists(task_ckpt_path):
+                os.mkdir(task_ckpt_path)
+        else:
+            ft_hparams.loggers = [wandb_logger,
+                                ObjectStoreLogger(object_store_cls=S3ObjectStore, object_store_kwargs={'bucket': bucket}, use_procs=False)]
+            ft_hparams.save_folder = f'{task}'
+            ft_hparams.save_num_checkpoints_to_keep = 0
+            save_artifact_name = f'{{run_name}}/{task}/ep{{epoch}}-ba{{batch}}-rank{{rank}}'
+            save_latest_artifact_name = f'{{run_name}}/{task}/latest-rank{{rank}}'
+            ft_hparams.save_artifact_name = save_artifact_name
+            ft_hparams.save_latest_artifact_name = save_latest_artifact_name
+            ft_hparams.save_overwrite = True
+
+    print(f" --------\n SPAWNING NEW TASK: {task}\n DEVICE: {torch.cuda.current_device()}\n --------")
+
+    trainer = ft_hparams.initialize_object()
+    trainer.fit(duration='3ep')
```

Comment:
Not sure we should hardcode the duration. I recommend to have the task YAMLs instead just use `max_duration: 3ep` so that it's at least more transparent where that duration comes from.

Also, are we sure all the fine-tuning tasks should run for 3 epochs? I seem to recall seeing task-specific durations.

Reply:
Sorry, this was just put in for testing purposes to shorten the amount of training required but will be fixed during cleanup. Thanks for pointing this out!!